minidns-dane-java7/src/main/java/de/measite/minidns/dane/DaneExtendedTrustManager.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2015 the original author or authors
+ *
+ * This software is licensed under the Apache License, Version 2.0,
+ * the GNU Lesser General Public License version 2 or later ("LGPL")
+ * and the WTFPL.
+ * You may choose either license to govern your use of this software only
+ * upon the condition that you accept all of the terms of either
+ * the Apache License 2.0, the LGPL 2.1+ or the WTFPL.
+ */
+package de.measite.minidns.dane;
+
+import de.measite.minidns.AbstractDNSClient;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509ExtendedTrustManager;
+import javax.net.ssl.X509TrustManager;
+import java.net.Socket;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.logging.Logger;
+
+public class DaneExtendedTrustManager extends X509ExtendedTrustManager {
+    private final static Logger LOGGER = Logger.getLogger(DaneExtendedTrustManager.class.getName());
+
+    private final X509TrustManager base;
+    private final DaneVerifier verifier;
+
+    public static void inject() {
+        inject(new DaneExtendedTrustManager());
+    }
+
+    public static void inject(DaneExtendedTrustManager trustManager) {
```

Comment:
Arghs, ok, that makes sense, although I'm not sure if it should be injected this way.....

I've yet to see a library that does not support changing the ssl context, and injecting it as a default might be a bad idea (changes s.th. for the whole JVM - and especially a corner piece)

For Android: It looks like this was added in 2014, yet it's '@hidden' and I can't find it in the source tree right now: https://android-review.googlesource.com/#/c/88585/ (Android offers many Java 7 APIs, and even Java 8 APIs in some places (see e.g. GZIPInput/OutputStream) yet it's source compatibility is mainly stuck at version 6 - the APIs have never been fully Java 6 compatible AFAIK)

But I could imagine that this would be useful even without full SNI compatibility (e.g. with a new trust manager per connection).

Anyway, makes sense.